src/Api.php

```diff
@@ -138,14 +151,13 @@ public function call($method, $path, array $data = [], array $headers = []) {
     }
 
     // Check to see if this request requires a version.
-    foreach ($resource->getVersionedTypes() as $type) {
-
-      if (strpos($request->getHeaderLine('accept'), $type) !== FALSE && !$request->getVersion()) {
-        return $this->toError(t('Missing required API version number.'), 'missing_version', 400);
-      }
-
+    $accept_header   = $request->getHeaderLine('Accept');
+    $versioned_types = $resource->getVersionedTypes();
+    if ($accept_header && $versioned_types && $this->negotiator->getBest($accept_header, $versioned_types) && !$request->getVersion()) {
```

Comment:
This is basically an improved version of #19 now that there's a negotiator involved. The negotiator will return a truthy value if the request accepts any of the content types requiring versioning.